File: internal/model/rebalance.go
package model

import "time"

// Rebalance 再平衡记录
type Rebalance struct {
	ID               uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	StockRatio       float64   `gorm:"not null" json:"stock_ratio"`        // 股票比例
	BondRatio        float64   `gorm:"not null" json:"bond_ratio"`         // 债券比例
	TotalAmount      float64   `gorm:"not null" json:"total_amount"`       // 总金额
	StockAmount      float64   `gorm:"not null" json:"stock_amount"`       // 股票金额
	BondAmount       float64   `gorm:"not null" json:"bond_amount"`        // 债券金额
	TargetStockRatio float64   `gorm:"not null" json:"target_stock_ratio"` // 目标股票比例
	TargetBondRatio  float64   `gorm:"not null" json:"target_bond_ratio"`  // 目标债券比例
	Note             string    `gorm:"type:text" json:"note"`              // 备注
	CreatedAt        time.Time `gorm:"autoCreateTime" json:"created_at"`
}

func (Rebalance) TableName() string {
	return "rebalances"
}

File: lib/data/models/rebalance.dart
/// 再平衡记录模型
class Rebalance {
  final int? id;
  final double stockRatio;
  final double bondRatio;
  final double totalAmount;
  final double stockAmount;
  final double bondAmount;
  final double targetStockRatio;
  final double targetBondRatio;
  final String? note;
  final DateTime createdAt;

  const Rebalance({
    this.id,
    required this.stockRatio,
    required this.bondRatio,
    required this.totalAmount,
    required this.stockAmount,
    required this.bondAmount,
    required this.targetStockRatio,
    required this.targetBondRatio,
    this.note,
    required this.createdAt,
  });

  /// 从数据库 Map 创建
  factory Rebalance.fromMap(Map<String, dynamic> map) {
    return Rebalance(
      id: map['id'] as int?,
      stockRatio: map['stock_ratio'] as double,
      bondRatio: map['bond_ratio'] as double,
      totalAmount: map['total_amount'] as double,
      stockAmount: map['stock_amount'] as double,
      bondAmount: map['bond_amount'] as double,
      targetStockRatio: map['target_stock_ratio'] as double,
      targetBondRatio: map['target_bond_ratio'] as double,
      note: map['note'] as String?,
      createdAt: DateTime.parse(map['created_at'] as String),
    );
  }

  /// 转换为数据库 Map
  Map<String, dynamic> toMap() {
    return {
      if (id != null) 'id': id,
      'stock_ratio': stockRatio,
      'bond_ratio': bondRatio,
      'total_amount': totalAmount,
      'stock_amount': stockAmount,
      'bond_amount': bondAmount,
      'target_stock_ratio': targetStockRatio,
      'target_bond_ratio': targetBondRatio,
      'note': note,
      'created_at': createdAt.toIso8601String(),
    };
  }
}
